backend/oasst_backend/prompt_repository.py

```diff
@@ -26,6 +26,24 @@ def __init__(self, db: Session, api_client: ApiClient, user: Optional[protocol_s
         self.user_id = self.user.id if self.user else None
         self.journal = JournalWriter(db, api_client, self.user)
 
+    def query_frontend_user(
+        self, auth_method: str, username: str, api_client_id: Optional[UUID] = None
+    ) -> Optional[User]:
+        if not self.api_client.trusted and not api_client_id:
+            api_client_id = self.api_client.id
+
+        if not self.api_client.trusted and api_client_id != self.api_client.id:
+            # Unprivileged API client asks for foreign user
+            raise OasstError("Forbidden", OasstErrorCode.API_CLIENT_NOT_AUTHORIZED, HTTP_403_FORBIDDEN)
+
+        user_query = self.db.query(User).filter(User.auth_method == auth_method, User.username == username)
+
+        if api_client_id:
+            user_query = user_query.filter(User.api_client_id == api_client_id)
+
+        user: User = user_query.first()
```

Comment:
is it ok that multiple users may match here? I'd guess this can happen if a trusted frontend supplies `api_client_id=None`. When would that happen? Why would it be interested in the first?

Reply:
The spec for the issue is that trusted clients should be able to resolve any user. I sort of assumed usernames were unique per auth method but maybe that's not the case? If not I agree it isn't clear we want the first, maybe would be better to return all matches in that case

Reply:
This now requires an `api_client_id` so avoiding this issue